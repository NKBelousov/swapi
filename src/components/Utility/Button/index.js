import React, { PureComponent } from "react";
import PropTypes from "prop-types";

import styled from "styled-components";

const Block = styled.div`
  background: ${props => props.theme.primary};
  border: 1px solid ${props => props.theme.primary};
  color: ${props => props.theme.secondary};
  cursor: pointer;
  display: block;
  margin: 0 auto;
  text-align: center;

  &:hover {
    background: ${props => props.theme.secondary};
    color: ${props => props.theme.primary};
  }

  &:active {
    box-shadow: 0 0 15px #000 inset;
  }
`;

const Text = styled.span`
  font-size: 2rem;
  padding: 1rem;
`;

class Button extends PureComponent {
  static get propTypes() {
    return {
      onClick: PropTypes.func.isRequired,
      text: PropTypes.string,
    };
  }
  static get defaultProps() {
    return {
      text: "Button",
    };
  }
  render() {
    return (
      <Block onClick={this.props.onClick}>
        <Text>{this.props.text}</Text>
      </Block>
    );
  }
}

export default Button;
