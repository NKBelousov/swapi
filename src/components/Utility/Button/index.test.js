import React from "react";
import renderer from "react-test-renderer";

import Button from "./index.js";

test("Should render correctly", () => {
  const text = "Hello, World!";
  const onClick = () => {
    alert(text);
  };
  const tree = renderer
    .create(<Button text={text} onClick={onClick} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
