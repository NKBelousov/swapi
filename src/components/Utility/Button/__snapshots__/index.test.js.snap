// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<div
  className="sc-bdVaJa jXnzlh"
  onClick={[Function]}
>
  <span
    className="sc-bwzfXH gYPdj"
  >
    Hello, World!
  </span>
</div>
`;
